halo/app/app.go

```diff
@@ -136,9 +142,19 @@ func newApp(
 		&app.EvidenceKeeper,
 		&app.UpgradeKeeper,
 		&app.SlashingEventProc,
-		&app.StakingEventProc,
 		&app.UpgradeEventProc,
-	); err != nil {
+	}
+
+	// TODO (christian): remove feature check with logs
+	if feature.FlagEVMStakingModule.Enabled(ctx) {
+		log.Info(ctx, "EVMStakingModule feature ENABLED")
+		dependencies = append(dependencies, &app.StakingEventKeeper)
+	} else {
+		log.Info(ctx, "EVMStakingModule feature DISABLED")
```

Comment:
No need to log if flags are disabled, this is default behaviour

Reply:
Yep that was only for my debugging to make sure we're going though this exact code path. Will remove.